src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -429,105 +405,137 @@ private async Task<EntityStateResponse<T>> ReadEntityStateAsync<T>(TaskHubClient
         /// <inheritdoc />
         async Task<PurgeHistoryResult> IDurableOrchestrationClient.PurgeInstanceHistoryAsync(string instanceId)
         {
-            // TODO this cast is to avoid to adding methods to the core IOrchestrationService/Client interface in DurableTask.Core. Eventually we will need
-            // a better way of handling this
-            var serviceClient = this.client.ServiceClient as AzureStorageOrchestrationService;
-            if (serviceClient == null)
-            {
-                throw new NotSupportedException("Only the Azure Storage state provider is currently supported for the purge instance history feature");
-            }
-
-            AzureStorage.PurgeHistoryResult purgeHistoryResult =
-                await serviceClient.PurgeInstanceHistoryAsync(instanceId);
-            return new PurgeHistoryResult(purgeHistoryResult.InstancesDeleted);
+            int numInstancesDeleted = await this.durableFunctionsClient.PurgeInstanceHistoryByInstanceId(instanceId);
```

Comment:
Related to another one of my comments, would it make more sense to have `PurgeInstanceHistoryByInstanceId` return an instance of `PurgeHistoryResult`?